graylog2-web-interface/src/pages/ConfigurationsPage.tsx

```diff
@@ -0,0 +1,240 @@
+  <Col md={6}>
+    <Boundary title={title}>
+      {children}
+    </Boundary>
+  </Col>
+);
+
+const _onUpdate = (configType: string) => {
+  return (config) => {
+    switch (configType) {
+      case MESSAGE_PROCESSORS_CONFIG:
+        return ConfigurationsActions.updateMessageProcessorsConfig(configType, config);
+      case URL_WHITELIST_CONFIG:
+        return ConfigurationsActions.updateWhitelist(configType, config);
+      default:
+        return ConfigurationsActions.update(configType, config);
+    }
+  };
+};
+
+const _getConfig = (configType, configuration) => configuration?.[configType] ?? null;
+
+const _pluginConfigs = (systemConfigs, configuration) => systemConfigs
+  .map(({ component: SystemConfigComponent, configType }) => (
+    <ConfigletContainer title={configType}>
+      <SystemConfigComponent key={`system-configuration-${configType}`}
+                             config={_getConfig(configType, configuration) ?? undefined}
+                             updateConfig={_onUpdate(configType)} />
+    </ConfigletContainer>
+  ));
+
+type PluginConfigRowsProps = {
+  configuration: Record<string, any>,
+  systemConfigs: Array<SystemConfiguration>,
+};
+
+const PluginConfigRows = ({ configuration, systemConfigs }: PluginConfigRowsProps) => {
+  const pluginConfigs = useMemo(() => _pluginConfigs(systemConfigs, configuration), [configuration, systemConfigs]);
+
+  // Put two plugin config components per row.
+  const configRows = chunk(pluginConfigs, 2)
+    .map((configChunk, idx) => (
+      // eslint-disable-next-line react/no-array-index-key
+      <Row key={`plugin-config-row-${idx}`}>
+        {configChunk[0]}
+        {configChunk[1]}
+      </Row>
+    ));
+
+  return <>{configRows}</>;
+};
+
+const ConfigurationsPage = () => {
+  const [loaded, setLoaded] = useState(false);
+  const pluginSystemConfigs = usePluginEntities('systemConfigurations');
+  const configuration = useStore(ConfigurationsStore as Store<Record<string, any>>, (state) => state?.configuration);
+  const permissions = useStore(CurrentUserStore as Store<{ currentUser: { permissions: Array<string> } }>, (state) => state?.currentUser?.permissions);
+
+  useEffect(() => {
+    style.use();
+
+    return () => { style.unuse(); };
+  }, []);
+
+  useEffect(() => {
+    const promises = [
+      ConfigurationsActions.list(SEARCHES_CLUSTER_CONFIG),
+      ConfigurationsActions.listMessageProcessorsConfig(MESSAGE_PROCESSORS_CONFIG),
+      ConfigurationsActions.list(SIDECAR_CONFIG),
+      ConfigurationsActions.list(EVENTS_CONFIG),
+    ];
+
+    if (isPermitted(permissions, ['urlwhitelist:read'])) {
+      promises.push(ConfigurationsActions.listWhiteListConfig(URL_WHITELIST_CONFIG));
+    }
+
+    const pluginPromises = pluginSystemConfigs
+      .map((systemConfig) => ConfigurationsActions.list(systemConfig.configType));
+
+    Promise.all([...promises, ...pluginPromises]).then(() => setLoaded(true));
```

Comment:
I like this change, way cleaner than before! However I see a problem here: when one of the configurations can't load (maybe there is some network issue or the server threw an exception), none of the configurations will be rendered and the page will stay in a loading state.

I think the easiest solution would be replacing `Promise.all` with [`Promise.allSettled`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/allSettled), and display an error message in those configurations that couldn't load. `Promise.allSettled` is quite recent but has already [~90% browser support](https://caniuse.com/mdn-javascript_builtins_promise_allsettled) and our `promise-polyfill` version should already support it. 

Alternatively we could let each configlet fetch its own configuration. That has some drawbacks on its own, but would decouple all fetches and would also make more explicit the relationship between each configuration and its data.

Reply:
Great finding! I checked that our polyfill supports `Promise.allSettled` and will use that instead. Thanks a lot for including a suggestion for a solution in your comment!